### PR TITLE
20732: implemented multi tier transit

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 1]
```diff
@@ -728,6 +734,14 @@ func resourceAviatrixTransitGatewayCreate(d *schema.ResourceData, meta interface
 			return fmt.Errorf("\"ha_oob_management_sbunet\" must be empty if \"enable_private_oob\" is false")
 		}
 	}
+
+	enableMultitierTransit := d.Get("enable_multitier_transit").(bool)
+	if enableMultitierTransit {
+		if d.Get("local_as_number") == "" {
+			return fmt.Errorf("local_as_number required to enalbe multi tier transit")
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_transit_gateway.go]: typo: enalbe -> enable

File: docs/resources/aviatrix_transit_gateway.md [hunk 2]
```diff
@@ -181,6 +181,7 @@ The following arguments are supported:
 * `local_as_number` - (Optional) Changes the Aviatrix Transit Gateway ASN number before you setup Aviatrix Transit Gateway connection configurations.
 * `bgp_ecmp` - (Optional) Enable Equal Cost Multi Path (ECMP) routing for the next hop. Default value: false.
 * `enable_bgp_over_lan` - (Optional) Pre-allocate a network interface(eth4) for "BGP over LAN" functionality. Must be enabled to create a BGP over LAN `aviatrix_transit_external_device_conn` resource with this Transit Gateway. Only valid for cloud_type = 8 (AZURE). Valid values: true or false. Default value: false. Available as of provider version R2.18+
+* `enable_multitier_transit` - (Optional) Enable Multi-tier Transit mode on transit gateway. When enabled, transit gateway will propagate routes it receives from its transit peering peer to other transit peering peers. `local_as_number` is required. Default value: false.
```
Comment (CyrusJavan) [inline docs/resources/aviatrix_transit_gateway.md]: Please append "Available as of provider version R2.19+"

File: aviatrix/data_source_aviatrix_transit_gateway.go [hunk 3]
```diff
@@ -256,6 +256,11 @@ func dataSourceAviatrixTransitGateway() *schema.Resource {
 				Computed:    true,
 				Description: "A map of tags assigned to the transit gateway.",
 			},
+			"enable_multitier_transit": {
```
Comment (CyrusJavan) [inline aviatrix/data_source_aviatrix_transit_gateway.go]: Should we add a underscore between multi and tier, like `enable_multi_tier_transit`? 

multitier can be hard to read if you don't know what it is supposed to be.

File: goaviatrix/transit_vpc.go [hunk 4]
```diff
@@ -70,6 +70,7 @@ type TransitGatewayAdvancedConfig struct {
 	PeerVnetId                        []string
 	BgpHoldTime                       int
 	EnableSummarizeCidrToTgw          bool
+	EnableMultitierTransit            bool
```
Comment (CyrusJavan) [inline goaviatrix/transit_vpc.go]: Do we need to add this to Advanded Config if we can already get this value from GetGateway/ `list_vpcs_summary` ?

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 5]
```diff
@@ -2478,6 +2500,23 @@ func resourceAviatrixTransitGatewayUpdate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if d.HasChange("enable_multitier_transit") {
+		if d.Get("enable_multitier_transit").(bool) {
+			if d.Get("local_as_number") == "" {
+				return fmt.Errorf("local_as_number required to enalbe multi tier transit")
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_transit_gateway.go]: typo: enalbe -> enable

File: goaviatrix/transit_vpc.go (hunk 4 above)
Comment (shuoliugit) [reply goaviatrix/transit_vpc.go]: no need, removed.

**Review by CyrusJavan — DISMISSED** [review]

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]
Resolve conflicts, please.

**Comment by shuoliugit** [comment]
> Resolve conflicts, please.

fixed

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**